src/app/gui/queryresults/queryresultsservice.js

```diff
@@ -1264,37 +1277,29 @@ proto.getVectorLayerFeaturesFromQueryRequest = function(vectorLayer, query={}) {
 
 };
 
-proto._addVectorLayersDataToQueryResponse = function() {
-  this.onbefore('setQueryResponse', (queryResponse, options = {}) => {
```

Comment:
@volterra79 I've slightly rewrote your bugfix in order to remove the need for the above event listener. Let me know if i did something wrong.